content/blog/generating-self-signed-certificates-using-cert-manager-for-kubernetes-in-hpe-greenlake-for-private-cloud-entreprise.md

```diff
@@ -0,0 +1,392 @@
+
+Here is one such sample Ingress YAML manifest file *ingress-nginx-selfsigned.yaml*:
+
+```shell
+$ cat ingress-nginx-selfsigned.yaml
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: nginx-ingress-selfsigned
+  annotations:
+    ingress.kubernetes.io/ssl-redirect: "true"
+    cert-manager.io/issuer: "cfe-selfsinged-issuer"
+spec:
+  ingressClassName: nginx
+  tls:
+  - hosts:
+    - nginx.example.com
+    secretName: cfe-tls-key-pair
+  rules:
+  - host: nginx.example.com
+    http:
+      paths:
+      - path: /
+        pathType: Prefix
+        backend:
+          service:
+            name: nginx-app
+            port:
+              number: 80
+```
+
+It assumes the Nginx Ingress controller is deployed in the cluster. It configures the TLS block with the hostname *'nginx.example.com'* and the generated K8s secret. 
+
+One benefit of this approach is that the sample Nginx application can be deployed in the cluster with the default service type *ClusterIP*, which provides internal connectivity and can solely be accessed from within the cluster. The Ingress controller will p﻿rovide external access and handle SSL by accessing the certificate in the cluster and route the traffic to the deployed Nginx application in the backend.  
+
+T﻿ype the following command to deploy the Ingress resource to the namespace *cfe-apps*: 
+
+```shell
+$ kubectl apply -f ingress-nginx-selfsigned.yaml -n cfe-apps
+ingress.networking.k8s.io/nginx-ingress-selfsigned created
+```
+
+A﻿fter deploying the Ingress using the above command, together with Nginx application deployment, to the namespace *cfe-apps*, you can validate the Ingress TLS using the browser. 
+
+S﻿tart the browser and type the URL *nginx.example.com*, it will be rediected over HTTPS with the warning message *'Your connection is not private'*: 
+
+![](/img/nginx-private.png)
+
+Y﻿ou can click *Not secure* and start the Certificate Viewer to check the TLS certificate before clicking *Proceed to nginx.example.com (unsafe)* to go to the Nginx page:
+
+![](/img/nginx-cert.png)
+
+### Conclusion
+
+This blog post described the steps to generate a self-signed certificate using cert-manager for K8s in HPE GreenLake for Private Cloud Enterprise. Self-signed certificates provide an easy way to prove your own identity for the applications deployed in K8s cluster. They are a good option for development and testing environments. However, self-signed certificates should not be used for production applications. For production use cases, you can try out cert-manager with [Lets Encrypt](https://letsencrypt.org/). You can refer to [cert-manager documentation](https://cert-manager.io/docs/) on how to use it with the type of *Let’s Encrypt* challenges, as well as other sources than *Let’s Encrypt*.  
```

Comment:
This is a good option for development and testing environments. However, because self-signed certificates are not trusted certificates, they should not be used for production applications. For production use cases, ...

Reply:
@GuopingJia, no further suggestions from my side. let's wait for Dale's editorial review and her editorial suggestions for changes. Thanks!